### PR TITLE
feat(cloudsql): update sample apps with Kubernetes Engine yaml files

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: cloudsql/mysql/database-sql/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    metadata:
+      labels:
+        app: gke-cloud-sql-app
+    spec:
+      serviceAccountName: <YOUR-KSA-NAME>
+      containers:
+      - name: gke-cloud-sql-app
+        # Replace $LOCATION with your Artifact Registry location (e.g., us-central1).
+        # Replace $YOUR_PROJECT_ID with your project ID.
+        image: $LOCATION-docker.pkg.dev/$YOUR_PROJECT_ID/gke-cloud-sql-repo/gke-cloud-sql-image:latest
```
Comment (enocom) [inline cloudsql/mysql/database-sql/deployment.yaml]: Why the difference in variable declarations, $LOCATION vs <LOCATION>?

File: cloudsql/mysql/database-sql/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        # This app listens on port 8080 for web traffic by default.
+        ports:
+        - containerPort: 8080
+        env:
+        - name: PORT
+          value: "8080"
+        - name: DB_HOST
+          value: "0.0.0.0"
```
Comment (enocom) [inline cloudsql/mysql/database-sql/deployment.yaml:28]: 127.0.0.1 is probably better here.

File: cloudsql/mysql/database-sql/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ 
+          # tcp should be set to the port the proxy should listen on
+          # and should match the DB_PORT value set above.
+          # Defaults: MySQL: 3306, Postgres: 5432, SQLServer: 1433
+          - "-instances=<INSTANCE_CONNECTION_NAME>=tcp:3306"
+        securityContext:
+          # The default Cloud SQL proxy image runs as the
+          # "nonroot" user and group (uid: 65532) by default.
+          runAsNonRoot: true
+# [END cloud_sql_mysql_databasesql_gke_quickstart_deployment]
```
Comment (enocom) [inline cloudsql/mysql/database-sql/deployment.yaml:66]: nit: add newline

File: cloudsql/mysql/database-sql/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# [START cloud_sql_mysql_databasesql_gke_quickstart_deployment]
```
Comment (enocom) [inline cloudsql/mysql/database-sql/deployment.yaml:1]: Do the region tags need to encompass the entire YAML file?

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 3 above)
Comment (enocom) [reply cloudsql/mysql/database-sql/deployment.yaml:66]: ditto below

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 2 above)
Comment (enocom) [reply cloudsql/mysql/database-sql/deployment.yaml:28]: ditto below

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 1 above)
Comment (enocom) [reply cloudsql/mysql/database-sql/deployment.yaml]: ditto below

**Review by enocom — COMMENTED** [review]
What if we moved this into https://github.com/GoogleCloudPlatform/cloudsql-proxy/tree/main/examples, perhaps under a "quickstart" directory? We could avoid the multiple PRs and the drift between samples.

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 1 above)
Comment (jsimonweb) [reply cloudsql/mysql/database-sql/deployment.yaml]: Updated all variable placeholders to use same format.

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 2 above)
Comment (jsimonweb) [reply cloudsql/mysql/database-sql/deployment.yaml]: Done.

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 3 above)
Comment (jsimonweb) [reply cloudsql/mysql/database-sql/deployment.yaml]: Done.

File: cloudsql/mysql/database-sql/deployment.yaml (hunk 4 above)
Comment (jsimonweb) [reply cloudsql/mysql/database-sql/deployment.yaml:1]: Not necessary but current placement is for full inclusion in documentation.

**Review by enocom — APPROVED** [review]

**Pull request merged**